feat: Add fallible try_get* fns

Gibson431 opened the pull request:
When using a pool in embedded applications, it may be important that the pool does not increase in size.

E.g. A pool of framebuffers for an LCD. Allocating a new frame buffer might consume unreasonable amounts of resources.

fereidani commented:
Good additions, Thank you!
Would you please rollback `pub fn get(&self) -> RefLocalGuard<P, T>` changes ,as this crate is performance related then we have to benchmark many different cases to see if try_get will get inlined in different scenarios or not.

Gibson431 committed:
feat: Add fallible try_get* fns

When using a pool in embedded applications, it may be important
that the pool does not increase in size.

E.g. A pool of framebuffers for an LCD. Allocating a new frame buffer
might consume unreasonable amounts of resources.

Gibson431 commented:
Ah yes definitely! I have updated the commit

fereidani commented:
Thank you for your contribution!
